src/app/templates/SwapForm/SwapFormInput/SlippageToleranceInput/SlippageToleranceInput.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { forwardRef, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import classNames from 'clsx';
+
+import AssetField from 'app/atoms/AssetField';
+
+import { SlippageTolerancePresetButton } from './SlippageTolerancePresetButton/SlippageTolerancePresetButton';
+
+interface Props {
+  name: string;
+  value?: number;
+  error?: boolean;
+  onChange: (newValue?: number) => void;
+}
+
+const SLIPPAGE_PRESETS = [0.5, 1, 3];
+
+export const SlippageToleranceInput = forwardRef<HTMLInputElement, Props>(({ name, value, error, onChange }, ref) => {
+  const [customPercentageValue, setCustomPercentageValue] = useState<number>();
+  const [inputWidth, setInputWidth] = useState(40);
```

Comment:
magic constant 40 and looks like its used in code below to calculate width of input in `useEffect`